packages/features/ee/managed-event-types/hooks/useLockedFieldsManager.tsx

```diff
@@ -1,27 +1,66 @@
 // eslint-disable-next-line no-restricted-imports
 import { get } from "lodash";
-import React from "react";
+import type { TFunction } from "next-i18next";
+import { useState } from "react";
+import type { Dispatch, SetStateAction } from "react";
+import type { UseFormReturn } from "react-hook-form";
 import type z from "zod";
 
+import type { FormValues } from "@calcom/features/eventtypes/lib/types";
 import type { Prisma } from "@calcom/prisma/client";
 import { SchedulingType } from "@calcom/prisma/enums";
 import type { _EventTypeModel } from "@calcom/prisma/zod/eventtype";
-import { Tooltip } from "@calcom/ui";
-import { Lock } from "@calcom/ui/components/icon";
+import { Tooltip, Badge, Switch } from "@calcom/ui";
+import { Lock, Unlock } from "@calcom/ui/components/icon";
 
-export const LockedIndicator = (label: string) => (
-  <Tooltip content={<>{label}</>}>
-    <div className="bg -mt-0.5 ml-1 inline-flex h-4 w-4 rounded-sm p-0.5">
-      <Lock className="text-subtle hover:text-muted h-3 w-3" />
-    </div>
-  </Tooltip>
-);
+export const LockedIndicator = (
+  isManagedEventType: boolean,
+  [isLocked, setIsLocked]: [boolean, Dispatch<SetStateAction<boolean>>],
+  t: TFunction,
+  fieldName: string,
+  setUnlockedFields: (fieldName: string, val: boolean | undefined) => void,
+  options = { simple: false }
+) => {
+  const stateText = t(isLocked ? "locked" : "unlocked");
+  const tooltipText = t(
+    `${isLocked ? "locked" : "unlocked"}_fields_${isManagedEventType ? "admin" : "member"}_description`
+  );
+  return (
+    <Tooltip content={<>{tooltipText}</>}>
+      <div className="inline">
+        <Badge variant={isLocked ? "gray" : "green"} className="ml-2 transform gap-1.5 p-1">
+          {!options.simple && (
+            <>
+              {isLocked ? (
+                <Lock className="text-subtle h-3 w-3" />
+              ) : (
+                <Unlock className="text-subtle h-3 w-3" />
+              )}
+              <span className="font-medium">{stateText}</span>
+            </>
+          )}
+          {isManagedEventType && (
+            <Switch
+              onCheckedChange={(enabled) => {
+                setIsLocked(enabled);
+                setUnlockedFields(fieldName, !enabled || undefined);
+              }}
+              checked={isLocked}
+              small={!options.simple}
+            />
+          )}
+        </Badge>
+      </div>
+    </Tooltip>
```

Comment:
New lock/unlock indicator with switch with the possibility to just show the switch for alerts using the `simple` flag